### PR TITLE
intellij-idea-ultimate-edition: update to 2024.1.2

### PR DESCRIPTION
**Pull request by piekay**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)

### CODE REVIEW

File: srcpkgs/intellij-idea-ultimate-edition/files/idea-ultimate.desktop
Comment (classabbyamp) [inline srcpkgs/intellij-idea-ultimate-edition/files/idea-ultimate.desktop:1]: we should not be vendoring these desktop files https://github.com/void-linux/void-packages/issues/49164

Comment (piekay) [reply srcpkgs/intellij-idea-ultimate-edition/files/idea-ultimate.desktop:1]: I get that, but I sadly don't see a way around for Jetbrains products, which don't offer a desktop entry by default, which is problematic for a GUI application

Comment (classabbyamp) [reply srcpkgs/intellij-idea-ultimate-edition/files/idea-ultimate.desktop:1]: upstream says to create the desktop file from within the application: https://www.jetbrains.com/help/idea/installation-guide.html#create-desktop-entry

Comment (classabbyamp) [reply srcpkgs/intellij-idea-ultimate-edition/files/idea-ultimate.desktop:1]: jars are archives, so we can do shenanigans:
```
bsdtar xOf lib/app-client.jar entry.desktop
[Desktop Entry]
Version=1.0
Type=Application
Name=$NAME$
Icon=$ICON$
Exec=$SCRIPT$ %f
Comment=$COMMENT$
Categories=Development;IDE;
Terminal=false
StartupWMClass=$WM_CLASS$
StartupNotify=true
```
Use this as a base. run the application to grab what it replaces the `$templates$` with.

e.g.:
```bash
post_extract() {
	bsdtar xf lib/app-client.jar entry.desktop
}

post_patch() {
	vsed -i -e 's/$NAME$/whatever/; s/$ICON$/whatever; ...' entry.desktop
}

do_install() {
	vinstall entry.desktop 644 usr/share/applications idea-ultimate.desktop
}
```

File: srcpkgs/intellij-idea-ultimate-edition/template [hunk 1]
```diff
@@ -1,90 +1,46 @@
...
-
-	ln -sf /${TARGET_PATH}/bin/idea.sh ${DESTDIR}/usr/bin/idea-ultimate
-	vdoc "${FILESDIR}/README.voidlinux"
+	vmkdir usr/share/doc/intellij-idea-ultimate
+	vmkdir usr/lib/intellij-idea-ultimate
+	vmkdir usr/share/pixmaps
+	vmkdir usr/share/icons/hicolor/scalable/apps
+
+	mv bin lib plugins product-info.json build.txt jbr ${DESTDIR}/usr/lib/intellij-idea-ultimate
+	mv license ${DESTDIR}/usr/share/doc/intellij-idea-ultimate
```
Comment (classabbyamp) [inline srcpkgs/intellij-idea-ultimate-edition/template:39]: use `vcopy` instead of `mv`

File: srcpkgs/intellij-idea-ultimate-edition/template [hunk 2]
```diff
@@ -1,90 +1,46 @@
...
+	rm ${DESTDIR}/usr/lib/intellij-idea-ultimate/lib/async-profiler -rf
+	rm ${DESTDIR}/usr/lib/intellij-idea-ultimate/lib/async-profiler.jar
+	ln -sf /usr/lib/intellij-idea-ultimate/bin/idea.sh ${DESTDIR}/usr/bin/idea-ultimate
+	ln -sf /usr/lib/intellij-idea-ultimate/bin/idea.png ${DESTDIR}/usr/share/pixmaps/idea-ultimate.svg
+	ln -sf /usr/lib/intellij-idea-ultimate/bin/idea.svg ${DESTDIR}/usr/share/icons/hicolor/scalable/apps/idea-ultimate.svg
```
Comment (classabbyamp) [inline srcpkgs/intellij-idea-ultimate-edition/template:44]:
```suggestion
	ln -sf ../lib/intellij-idea-ultimate/bin/idea.sh ${DESTDIR}/usr/bin/idea-ultimate
	ln -sf ../lib/intellij-idea-ultimate/bin/idea.png ${DESTDIR}/usr/share/pixmaps/idea-ultimate.svg
	ln -sf ../lib/intellij-idea-ultimate/bin/idea.svg ${DESTDIR}/usr/share/icons/hicolor/scalable/apps/idea-ultimate.svg
```
relative symlinks are better

File: srcpkgs/intellij-idea-ultimate-edition/template [hunk 3]
```diff
@@ -1,90 +1,46 @@
...
 pkgname=intellij-idea-ultimate-edition
-version=2021.3.3
-revision=2
-archs="i686 x86_64"
-depends="giflib libXtst jetbrains-jdk-bin lldb-devel"
-short_desc="Most intelligent Java IDE"
+version=2024.1.2
+revision=1
+archs="x86_64"
+depends="virtual?java-environment giflib libXtst hicolor-icon-theme"
```
Comment (classabbyamp) [inline srcpkgs/intellij-idea-ultimate-edition/template:6]: does this work with any jdk (8, 11, 17) or does it need at least some version?

Comment (piekay) [reply srcpkgs/intellij-idea-ultimate-edition/template:6]: Jetbrains recommends JDK17, but it should work with any supported version

**Comment by piekay** [comment]
Added a vlicense and removed all spaces

**Comment by piekay** [comment]
Commit naming scheme should now follow the CONTRIBUTING.md guidelines

**Pull request merged**